src/Utils/request/api-request.ts

```diff
@@ -0,0 +1,60 @@
+import careConfig from "@careConfig";
+
+import { Route } from "@/Utils/request/types";
+import { makeHeaders, makeUrl } from "@/Utils/request/utils";
+
+import { getResponseBody } from "./request";
+
+interface QueryOptions {
+  pathParams?: Record<string, string>;
+  queryParams?: Record<string, string>;
+  silent?: boolean;
+}
+
+async function queryRequest<TData, TBody>(
+  { path, method, noAuth }: Route<TData, TBody>,
+  options?: QueryOptions & { signal?: AbortSignal },
+): Promise<TData> {
```

Comment:
Can we also include support for passing request bodies too? Some APIs such as `/api/v1/valueset/{slug}/expand/` used for searching valuesets require a request body containing the search text and count. 

Seen this as a standard practice in couple of search APIs where search parameters can get really complicated that might exceed the URL's query params limits.

Few references:

- https://softwareengineering.stackexchange.com/a/324158/415125
- https://docs.aws.amazon.com/healthlake/latest/devguide/search-with-post.html

Reply:
Also, why not keep `signal` attribute defined in the QueryOptions interface itself?